library/ssl_srv.c

```diff
@@ -2847,6 +2847,7 @@ static int ssl_write_server_hello( mbedtls_ssl_context *ssl )
     {
         MBEDTLS_SSL_DEBUG_MSG( 3, ( "session successfully restored from cache" ) );
         ssl->handshake->resume = 1;
+        ssl->handshake->peer_authenticated = MBEDTLS_SSL_FI_FLAG_SET;
```

Comment:
I'm not sure I'm a big fan of that. Just because we're resuming a session doesn't mean the peer had been authenticated last time. Can we instead not set the flag here, and later only check its value for non-resumed handshakes?

Reply:
I prefer not to have too complex logic in the end, there is a collected setting of this flag now in the `mbedtls_ssl_handshake_wrapup` if the `ssl->handshake->resume == 1` before possibly cleaning up the handshake-struct.